### DISCUSSION TITLE
IBM Semeru: jdk18 is eol and jdk19 is not yet released. Can we make 17 default?

### DISCUSSION THREAD

**Discussion by AdamBrousseau** [Homebrew/discussions#3897] [Casks] [answered]
Hi,

I am one of the release engineers for IBM Semeru Runtimes. We currently have 4 casks for Semeru[1][2][3][4]. jdk18 is the latest (18.0.2) and we have 3 LTS releases in cask-versions (jdk8, 11 and 17). 18 is now eol/eos however, we have not yet released jdk19 and there is no public eta at this point. We just had a release that updated the versions of 8, 11, and 17. Given 18.0.2 now has published security vulnerabilities, I think we should make the default 17 (17.0.5), remove 18, and keep 8 and 11 going as is. Once jdk19 is released we can update to something more similar to what is in place now.

The one caveat is our AArch64 binary, which we release on jdk11+ (no jdk8 yet).  Up until this release, it was marked as early access (ea). For this release, it is planned to become ga. However, as a result, there is a bit of delay releasing this platform so the binaries are not yet available. Possible eta is late November. I was holding off adding AArch64 to the casks because it was ea [5] but I see it was added recently [6][7]. Therefore, I don't believe the 11 and 17 casks can be updated until AArch64 is available.

Please advise how you would like to proceed.
Thanks,
Adam

[1] https://github.com/Homebrew/homebrew-cask/blob/master/Casks/semeru-jdk-open.rb
[2] https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/semeru-jdk8-open.rb
[3] https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/semeru-jdk11-open.rb
[4] https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/semeru-jdk17-open.rb
[5] https://github.com/Homebrew/homebrew-cask/pull/130231
[6] https://github.com/Homebrew/homebrew-cask-versions/pull/15046
[7] https://github.com/Homebrew/homebrew-cask/pull/134710

**Comment by SMillerDev** [comment, marked as answer]
I think the best course of action is to mark the cask discontinued (because it is) and reactivate it as soon as 19 is out. That way people know about the support state and can take appropriate action.

**Reply by AdamBrousseau** [reply]
Does that mean there will be no Semeru available unless users have the extended versions enabled?

**Reply by SMillerDev** [reply]
People will be able to install it, it will just come with a big warning that the version they installed is discontinued. They can then choose to find another version in `cask-versions` or another JDK.

**Reply by AdamBrousseau** [reply]
Thanks, Sean.

**Reply by AdamBrousseau** [reply]
https://github.com/Homebrew/homebrew-cask/pull/135466

**Comment by AdamBrousseau** [comment]
@SMillerDev is there a Homebrew policy around which version is the default or is this up to the product?
We (Semeru) discussed this yesterday and our preference would be to make the latest LTS (currently jdk17) as the default and offer the other LTS versions and the latest feature release (18, 19 etc) in the versions repository.

**Reply by SMillerDev** [reply]
The default is the latest version.
